Marlin/src/HAL/HAL_DUE/Tone.cpp

```diff
@@ -60,7 +53,6 @@ HAL_TONE_TIMER_ISR {
     digitalWrite(tone_pin, (pin_state ^= 1));
   }
   else noTone(tone_pin);                         // turn off interrupt
-  HAL_timer_restrain_count(TONE_TIMER_NUM, 10);  // make sure next ISR isn't delayed by up to 2 minutes
 }
 
 #endif // ARDUINO_ARCH_SAM
```

Comment:
I've verified duration & frequency are correct via a logic analyzer.

Reply:
Sorry, @Bob-the-Kuhn ... Yesterday i was too tired, and fell asleep... (was 2AM ...I am at GMT-3).
I do agree with all your comments.
I will review the timer code against the datasheet 👍 
It is not the calculation of the frequency the problem. I think the problem is to alter the period while the timer is running. That is why I added an TC_Stop() in the HAL_timers.cpp:HAL_timer_start

Reply:
If you take a look at my Tone.cpp file, you will notice we agree :+1: ... I will remove the line
tc->TC_CHANNEL[channel].TC_CV = 0;

Reply:
And all the unused functions from HAL_timers.h

Reply:
BTW, hw reviews are always hard. Dont worry: The usual "I like it/I dislike it/do it other way, this does not make sense"... It´s fine with me 👍

Reply:
TC_Configure stops the clock to the counter so TC_Stop() isn't need.  As far as I can tell none of the changes in that file are required but none of them hurt either.  Your choice.

Reply:
.platformio\packages\framework-arduinosam\system\sam\libsam\source\tc.c is where all the low level TC routines are at.